src/Common/Internal/Http/HttpClient.php

```diff
@@ -279,23 +273,38 @@ public function send($filters, $url = null)
             $this->_request = $filter->handleRequest($this)->_request;
         }
 
-        $this->_response = $this->_request->send();
+        $response = $this->_request->send();
 
         $start = count($filters) - 1;
         for ($index = $start; $index >= 0; --$index) {
-            $this->_response = $filters[$index]->handleResponse(
-                $this, $this->_response
-            );
+            $response = $filters[$index]->handleResponse($this, $response);
         }
 
         self::throwIfError(
-            $this->_response->getStatus(),
-            $this->_response->getReasonPhrase(),
-            $this->_response->getBody(),
+            $response->getStatus(),
+            $response->getReasonPhrase(),
+            $response->getBody(),
             $this->_expectedStatusCodes
         );
 
-        return $this->_response->getBody();
+        return $response;
+    }
+
+    /**
+     * Processes the reuqest through HTTP pipeline with passed $filters,
+     * sends HTTP request to the wire and process the response in the HTTP pipeline.
+     *
+     * @param array $filters HTTP filters which will be applied to the request before
+     *                       send and then applied to the response.
+     * @param IUrl  $url     Request url.
+     *
+     * @throws WindowsAzure\Common\ServiceException
+     *
+     * @return string The response body
+     */
+    public function send($filters, $url = null)
+    {
+        return $this->sendAndGetResponse($filters, $url)->getBody();
```

Comment:
It shouldn't. The sendAndGetResponse() should throw an exception if there is any problem.